azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/extensions/AzureComputeImageExtension.java

```diff
@@ -123,6 +123,8 @@ public Image call() throws Exception {
 
             Image image = resourceDefinitionToImage.create(cloneTemplate.getSourceNodeId(), cloneTemplate.getName())
                   .apply(definitions.get(0));
+            checkState(image != null, "Image %s was not created for the requested node",
+                  cloneTemplate.getSourceNodeId());
```

Comment:
The log parameter is the id of the node. Better use a message like "Image *for node %s*..."